src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { EntryType } from 'app/shared/enum/entry-type';
+import { EntryUpdateTime, User, UsersService } from 'app/shared/openapi';
+import { UserQuery } from 'app/shared/user/user.query';
+import { debounceTime, finalize, takeUntil } from 'rxjs/operators';
+import { MyWorkflowsService } from 'app/myworkflows/myworkflows.service';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+
+@Component({
+  selector: 'app-entry-box',
+  templateUrl: './entry-box.component.html',
+  styleUrls: ['./entry-box.component.scss'],
+})
+export class EntryBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  @Input() entryType: EntryUpdateTime.EntryTypeEnum;
+  entryTypeLowerCase: string;
+  filterText: string;
+  listOfEntries: Array<EntryUpdateTime> = [];
+  user: User;
+  helpLink: string;
+  allEntriesLink: string;
+  hasEntries: boolean;
+  firstLoad = true;
+  public isLoading = true;
+
+  constructor(
+    private registerToolService: RegisterToolService,
+    private userQuery: UserQuery,
+    private myWorkflowsService: MyWorkflowsService,
+    private usersService: UsersService
+  ) {
+    super();
+    this.user = this.userQuery.getValue().user;
+  }
+
+  ngOnInit(): void {
+    this.getMyEntries();
+
+    if (this.entryType) {
+      this.entryTypeLowerCase = this.entryType.toLowerCase();
+    }
+
+    //Get the links for the specified entryType
+    if (this.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-workflows.html';
+      this.allEntriesLink = '/my-workflows/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-tools.html';
+      this.allEntriesLink = '/my-tools/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.SERVICE) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/getting-started-with-services.html';
+      this.allEntriesLink = '/my-services/';
+    }
+  }
+
+  getMyEntries() {
+    this.usersService
+      .getUserEntries(null, this.filterText)
+      .pipe(
+        finalize(() => (this.isLoading = false)),
+        debounceTime(750),
+        takeUntil(this.ngUnsubscribe)
+      )
+      .subscribe((myEntries: Array<EntryUpdateTime>) => {
+        this.listOfEntries = [];
+        myEntries
+          .filter(
+            (entry) =>
+              entry.entryType === this.entryType ||
+              (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL && entry.entryType === EntryUpdateTime.EntryTypeEnum.APPTOOL)
+          )
+          .forEach((entry: EntryUpdateTime) => {
+            if (this.listOfEntries.length < 7) {
```

Comment:
To followup, in addition to simplification, you're also going through the loop n times, e.g., 1,000 times if there are 1,000 items in the array, even though you only need the first 7. For the amount of data we're currently dealing with, it probably won't matter, performance-wise, but just as a best practice...

Reply:
Hmm that is true I originally did it to count the number of total entries but there should be a better endpoint for that.